src/cli/cli.cpp

```diff
@@ -708,6 +719,90 @@ void Interpreter::ProcessDiscover(int argc, char *argv[])
     AppendResult(error);
 }
 
+#if OPENTHREAD_ENABLE_DNS_CLIENT
+void Interpreter::ProcessDns(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    long port = OT_DNS_DEFAULT_SERVER_PORT;
+    Ip6::MessageInfo messageInfo;
+    otDnsQuery query;
+
+    if (argc && strcmp(argv[0], "resolve") == 0)
+    {
+        VerifyOrExit(!mResolvingInProgress, error = kThreadError_Busy);
+        VerifyOrExit(argc >= 2, error = kThreadError_Parse);
```

Comment:
Would `kThreadError_InvalidArgs` be more appropriate here, since the user would appear to be passing invalid args to the command line?